src/Designer/backend/src/Designer/Services/Implementation/GitOps/GitRepoGitOpsConfigurationManager.cs

```diff
@@ -0,0 +1,219 @@
+        await EnsureRemoteRepositoryExists(context);
+
+        await sourceControl.CloneRemoteRepository(gitOpsSettings.GitOpsOrg, GitOpsRepoName(context.Org));
+
+        await EnsureBaseManifests(context);
+        await EnsureEnvironmentManifests(context, environment);
+    }
+
+    private async Task EnsureRemoteRepositoryExists(AltinnOrgEditingContext context)
+    {
+        // Check if remote repo exists
+        var repo = await giteaApi.GetRepository(gitOpsSettings.GitOpsOrg, GitOpsRepoName(context.Org));
+        if (repo is not null)
+        {
+            return;
+        }
+
+        // Create repo with template
+        var createOptions = new CreateRepoOption(
+            name: GitOpsRepoName(context.Org),
+            description: $"GitOps configuration for {context.Org}",
+            autoInit: true,
+            makePrivate: false
+        );
+
+        await giteaApi.CreateRepository(gitOpsSettings.GitOpsOrg, createOptions);
+    }
+
+    private async Task EnsureBaseManifests(AltinnOrgEditingContext context)
+    {
+        var repository =
+            gitRepositoryFactory.GetAltinnGitRepository(gitOpsSettings.GitOpsOrg, GitOpsRepoName(context.Org), context.Developer);
+
+        if (!repository.FileExistsByRelativePath(ManifestsPathHelper.BaseManifests.KustomizationPath))
+        {
+            var baseManifests = gitOpsManifestsRenderer.GetBaseManifests();
+            await WriteManifestsToFiles(context, baseManifests);
+        }
+    }
+
+    private async Task EnsureEnvironmentManifests(AltinnOrgEditingContext context, AltinnEnvironment environment)
+    {
+        var repository =
+            gitRepositoryFactory.GetAltinnGitRepository(gitOpsSettings.GitOpsOrg, GitOpsRepoName(context.Org), context.Developer);
+
+        if (!repository.FileExistsByRelativePath(ManifestsPathHelper.EnvironmentManifests.KustomizationPath(environment.Name)))
+        {
+            var envManifests = gitOpsManifestsRenderer.GetEnvironmentOverlayManifests(environment, new HashSet<AltinnRepoName>());
+            await WriteManifestsToFiles(context, envManifests);
+        }
+    }
+
+    private void DeleteLocalRepositoryIfExists(AltinnOrgEditingContext context)
+    {
+        // Fast rename instead of deletion
+        // Should check if repo actually exists. GetAltinnGitRepository fails if do not exists
+        string repoPath = serviceRepositorySettings.GetServicePath(gitOpsSettings.GitOpsOrg, GitOpsRepoName(context.Org), context.Developer);
+        if (!Directory.Exists(repoPath))
+        {
+            return;
+        }
+
+        string renamedPath = $"{repoPath}-{timeProvider.GetUtcNow():yyyyMMddHHmmss}";
+        Directory.Move(repoPath, renamedPath);
+
+        // Fire and forget deletion task
+        _ = Task.Run(() =>
+        {
+            try
+            {
+                Directory.Delete(renamedPath, true);
+            }
+            catch
+            {
+                // Ignore deletion errors
```

Comment:
Do we actually not care if we fail here? Not even going to log a warning? Completely unfamiliar with this code here so just asking. Was just thinking if we have a bug in the try block here and it actually _never_ works, we would end up with a bunch of folders taking space and never getting cleaned up or something?

Reply:
Added logging. 
If deletion fails cleanup should be solved with https://github.com/Altinn/altinn-studio/issues/16445